Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp

```diff
@@ -282,6 +283,12 @@ static Fetch::Infrastructure::BodyWithType extract_body(XMLHttpRequestBodyInit c
             if (!blob->type().is_empty())
                 type = blob->type().to_byte_buffer();
         },
+        [&](JS::Handle<JS::Object> const& buffer_source) {
+            // Set source to a copy of the bytes held by object.
+            auto data_buffer = Bindings::IDL::get_buffer_source_copy(*buffer_source.cell());
+            if (data_buffer.has_value())
+                source = data_buffer.release_value();
```

Comment:
What about the error case? We can't just succeed if this fails. (related, `get_buffer_source_copy` should return `ErrorOr<ByteBuffer>`, not `Optional<ByteBuffer>`)